flow_screen_components/quickChoiceFSCproject/force-app/main/default/lwc/quickChoiceFSC/quickChoiceFSC.js

```diff
@@ -1,167 +1,215 @@
-                        console.log('options is: ' + options);
-                        index += 1;
-                    });
-                break;
-
-                default:
-                
-
-            }
-            this.options = options;
-        }
-            
-        else {
-            console.log('SmartChoiceFSC: Need a valid Input Mode value. Didn\'t get one');
-            throw new Error('SmartChoiceFSC: Need a valid Input Mode value. Didn\'t get one');
-        }
-    
-
-     }
-
-    @api
-    validate() {
-        let errorMessage = 'You must make a selection in: ' +  this.masterLabel + ' to continue'; 
-        
-        if((this.required === true) && (this.value === 'None')) { 
-           //If the component is invalid, return the isValid parameter as false and return an error message. 
-           return { 
-                isValid: false, 
-                errorMessage: errorMessage,
-            }; 
-        } 
-        else {   
-            return { isValid: true }; 
-        }
-    }
-     
-    
-
-    handleChange(event) {
-        this.selectedValue = event.detail.value;
-        console.log('selected value is: ' + this.selectedValue);
-        const attributeChangeEvent = new FlowAttributeChangeEvent('value', this.selectedValue);
-        this.dispatchEvent(attributeChangeEvent);
-    }
-}
\ No newline at end of file
+	@api value;
+	@api masterLabel;
+	@api choiceLabels;
+	@api choiceValues; //string collection
+	@api choiceIcons; 
+	@api includeIcons;
+	@api iconSize;
+	@api displayMode;
+	@api allowNoneToBeChosen;
+
+	@api recordTypeId; //used for picklist fields
+
+	@api objectName; //used for picklist fields
+	@api fieldName; //used for picklist fields
+	masterRecordTypeId = "012000000000000AAA"; //if a recordTypeId is not provided, use this one
+	@api objectAndFieldName;
+
+	@api inputMode;
+	@api required;
+	picklistOptionsStorage;
+
+	@track selectedValue;
+	@track showRadio = true;
+	@track showVisual = false;
+	@track legitInputModes = [
+		"Picklist Field",
+		"Single String Collection",
+		"Dual String Collections",
+		"Visual Text Box"
+	];
+	@track options = [];
+	@track items = [];
+
+	//possibility master record type only works if there aren't other record types?
+	@wire(getPicklistValues, {
+		recordTypeId: "$recordTypeId",
+		fieldApiName: "$objectAndFieldName"
+	})
+	picklistValues({ error, data }) {
+		if (data) {
+			console.log("inside picklist wiring");
+
+			let picklistOptions = [];
+			if (this.allowNoneToBeChosen) 
+				picklistOptions.push({ label: "--None--", value: "None" });
+
+			// Picklist values
+			data.values.forEach(key => {
+				picklistOptions.push({
+					label: key.label,
+					value: key.value
+				});
+			});
+
+			this.picklistOptionsStorage = picklistOptions;
+			console.log("displayMode is" + this.displayMode);
+
+			// Visual Picker Selection
+			if (this.displayMode === "Visual") {
+				this.showVisual = true;
```

Comment:
This boolean approach isn't really scalable and wasn't a good initial design decision. can we just remove it? the existing displayMode should be sufficient, I suspect.

Reply:
showVisual and showRadio are conditional properties used in the template.  They are set based on the displayMode attibute.  I don't think a boolean expression is valid in the html template.

I did, however, remove the redundant setting of these in getPicklistValues and the uneccessary clearing of showRadio when displayMode is "Visual".

Reply:
I see what you mean.